substrate/internal/defs

```diff
@@ -37,14 +37,24 @@
 #	define SUBSTRATE_NO_DISCARD(...) __VA_ARGS__
 #endif
 
-#if __has_cpp_attribute(fallthrough) ||  __cplusplus >= 201402L
+#if __has_cpp_attribute(fallthrough) || __cplusplus >= 201603L
```

Comment:
This is to over prevent clang-eagerly telling us we can use fallthrough and it turning into a warning, right?

Reply:
No, [[fallthrough]] is actually C++17-- `201402L` is C++14, and that throws the following warning on MSVC:

```
C:\Users\Amalia\Desktop\Rachel\substrate\substrate/crypto/twofish(107): warning C5051: attribute 'fallthrough' requires at least '/std:c++17'; ignored
```